app/src/main/java/com/keylesspalace/tusky/util/SaveTootHelper.java

```diff
@@ -0,0 +1,188 @@
+import android.text.TextUtils;
+import android.util.Log;
+import android.webkit.MimeTypeMap;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.keylesspalace.tusky.BuildConfig;
+import com.keylesspalace.tusky.ComposeActivity;
+import com.keylesspalace.tusky.db.TootDao;
+import com.keylesspalace.tusky.db.TootEntity;
+import com.keylesspalace.tusky.entity.Status;
+
+import java.io.File;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+public class SaveTootHelper {
+
+    public static final String TAG = "SaveTootHelper";
+
+    private TootDao tootDao;
+    private Context context;
+
+    public SaveTootHelper(@NonNull TootDao tootDao, @NonNull Context context) {
+        this.tootDao = tootDao;
+        this.context = context;
+    }
+
+
+    @SuppressLint("StaticFieldLeak")
+    public boolean saveToot(@NonNull String content,
+                             @NonNull String contentWarning,
+                             @Nullable String savedJsonUrls,
+                             @NonNull List<ComposeActivity.QueuedMedia> mediaQueued,
+                             int savedTootUid,
+                             @Nullable String inReplyToId,
+                             @Nullable String replyingStatusContent,
+                             @Nullable String replyingStatusAuthorUsername,
+                             @NonNull Status.Visibility statusVisibility) {
+
+        if (TextUtils.isEmpty(content) && mediaQueued.isEmpty()) {
+            return false;
+        }
+
+        // Get any existing file's URIs.
+        ArrayList<String> existingUris = null;
+        if (!TextUtils.isEmpty(savedJsonUrls)) {
+            existingUris = new Gson().fromJson(savedJsonUrls,
+                    new TypeToken<ArrayList<String>>() {
+                    }.getType());
+        }
+
+        String mediaUrlsSerialized = null;
+        if (!ListUtils.isEmpty(mediaQueued)) {
+            List<String> savedList = saveMedia(mediaQueued, existingUris);
+            if (!ListUtils.isEmpty(savedList)) {
+                mediaUrlsSerialized = new Gson().toJson(savedList);
+                if (!ListUtils.isEmpty(existingUris)) {
+                    deleteMedia(setDifference(existingUris, savedList));
+                }
+            } else {
+                return false;
+            }
+        } else if (!ListUtils.isEmpty(existingUris)) {
+            /* If there were URIs in the previous draft, but they've now been removed, those files
+             * can be deleted. */
+            deleteMedia(existingUris);
+        }
+        final TootEntity toot = new TootEntity(savedTootUid, content, mediaUrlsSerialized, contentWarning,
+                inReplyToId,
+                replyingStatusContent,
+                replyingStatusAuthorUsername,
+                statusVisibility);
+
+        new AsyncTask<Void, Void, Void>() {
```

Comment:
I don't use AsyncTask much so I'm not sure if we have to execute them in a special way to make them parallel...
To silence the warning we should make a static class and pass DAO to it, right?